### PR TITLE
Save log to S3

### PR DESCRIPTION
**Pull request by gabriel**
Saves log to S3 (last 100kb bytes). The URL is public but is not
discoverable.

### CODE REVIEW

**Comment by gabriel** [comment]
@cjb @oconnor663

File: update/util.go [hunk 1]
```diff
@@ -83,3 +85,24 @@ func RemoveNilErrors(errs []error) []error {
...
 	return r
 }
+
+// RandString returns random (base32) string with prefix
+func RandString(numBytes int) (string, error) {
+	buf, err := RandBytes(numBytes)
+	if err != nil {
+		return "", err
+	}
+	str := base32.StdEncoding.EncodeToString(buf)
```
Comment (oconnor663) [inline update/util.go]: I'm not sure if `EncodeToString` does exactly what we need here. The result of encoding a single zero byte is `AE======`. We probably don't want the padding, and also the final character isn't fully random. I don't think this is a problem if we stick to multiples of 5 in the length, and it looks like you're using 20 above, but I'd worry about changing that length in the future.

**Comment by oconnor663** [comment]
Is the goal to be able to post these links to slack after a build? That would be the same as just uploading a snippet from a privacy perspective, but less likely to get truncated by Slack bugs/limits. Do I have that right?

**Comment by gabriel** [comment]
yeah also we can clear out the logs periodically so they stop working, as long as slack isn't ingesting these links :(

**Comment by oconnor663** [comment]
The only real downside I can think of is that this means the logs aren't searchable with slack search. Thoughts @cjb?

File: update/util.go (hunk 1 above)
Comment (gabriel) [inline update/util.go]: Increased to 32 bytes, changed func name to RandomID and removed padding chars

**Pull request merged**